packages/v4/src/content/design-guidelines/components/menu/menu.md

```diff
@@ -0,0 +1,82 @@
+
+### Grouped menus
+You may decide to group menu items to associate related items and/or to indicate a hierarchy of items. Items may be grouped using group headings, separators, or both.
+
+<img src="./img/grouped-menu.png" alt="grouped menu" width="348"/>
+
+1. **Group heading (optional):** Add a group heading when you want to name the group.
+
+2. **Separator:** Separators are horizontal dividers that help to group the menu items by clearly showing where one group ends and the next begins.
+
+### Multi-level fly-out menus
+Fly-out menus can be used when you want to expose sub-items from a parent node. PatternFly supports a single level fly-out, only.
+
+<img src="./img/flyout-menu.png" alt="flyout menu" width="348"/>
+
+If a menu item has sub-items, a “>” character will be shown to the right of the item label. When hovering over the item, the secondary menu will be exposed.
+
+### Filtering menu items
+If the list of possible items is very long, a filter may be added to make it easier for the user to find what they are looking for.
+
+<img src="./img/menu-filtering.png" alt="grouped menu" width="348"/>
+
+When filtering is being used, a search input will be added to the top of the menu and the list of items will be filtered as the user types.
+
+### Favoriting and other actions
+Favoriting or any other optional action can be associated with any menu item. In this case, making an item a favorite will duplicate in at the top of the menu. This may be useful when you have a long list of possible actions and want to “pin” a subset of items to the top of the menu.
+
+<img src="./img/favorities-menu.png" alt="menu with favorites" width="348"/>
+
+
+Any action that can be represented as an icon button can be placed in one or more menu items. In this case the favorites action is used to mark an item as a favorite.
+
+## Usage
+
+| **When you want to...** |                     |
+| ------------------- | ------------------- |
+| Expose a list of actions in a dropdown menu | Use an actions menu with actions and/or links. Add icons in front of menu items only if they will help to make items more recognizable. Include descriptive text for menu items that may require further definition. |
+| Create logical groupings of menu items... | Use separators to group items. Include group headings if you don’t feel like the meaning of the groups will be obvious. |
+| Expose a list of nested actions... | Use a fly-out menu. Any optional attributes of basic menus, including icons, groups, and descriptions can also be applied to flyout menus.|
+| Select from a list of options in a dropdown menu ... | Use an options select menu for selecting between values rather than actions. Icons, descriptions, and groupings can be used with option select menus to improve recognition or create logical option groups. |
```

Comment:
This last case is a little confusing, are you talking about select lists, or drop downs? It seems like you are talking about select lists here so I would probably remove any mention of "drop downs" to prevent confusion!

Would rephrase to "select from a list of options in a select menu..." "Use an options select menu etc"

Reply:
This one I'm not sure about. I was using the word "dropdown" to refer to the type of menu generically rather than the component named "Dropdown." Options select menus are used in the Options menu component as well as the Select component. Ultimately we are looking to use the Menu component as part of a composable dropdown menu approach where you could put together custom dropdown menus. What if I just get rid of the work "dropdown" here and just say, "Select from a list of options?"

Reply:
Yes, I think that would be a good approach. I just want to steer clear of using the word "dropdown" unless it's referring to the actual dropdown component